// manager/CommentManager.js
var CommentManager = function (data) {  
    this.data = data;
}

var request = require('request');
var util = require('util');

CommentManager.prototype.data = {}

CommentManager.handleData = function (data) {
  data = JSON.parse(data);
  
  var results = [];
  if (data && data.length > 1 && data[0].data && data[1].data && data[1].data.children.length > 0){
      console.log(data[1].data.children.length);
    for(var i = 0; i < data[0].data.children.length; i++){
      var result = data[1].data.children[i].data;
      if (result){
        results.push({
          date : result.created ? new Date(result.created).toString() : 'NaN',
          author: result.author,
          postId: data[0].data.children[0].data.id, //post
          commentContent: result.body,
          commentId : result.id
        });
      }
    }
  }

  return results;
}

CommentManager.findByPostId = function (postId, limit, offset, callback) {
console.log("https://www.reddit.com/r/personalfinance/comments/"+postId+".json?limit="+limit+"&offset=" + offset);
  request("https://www.reddit.com/r/personalfinance/comments/"+postId+".json?limit="+limit+"&offset=" + offset, function (error, response, body) {
    if (!error && response.statusCode == 200) {
      //console.log() // Show the HTML for the Google homepage.
      console.log("request comments ok");
      callback(null, CommentManager.handleData(response.body));
    }
    else
      callback(error, response);
  });
}

module.exports = CommentManager;
