// manager/PostManager.js
var PostManager = function (data) {  
    this.data = data;
}

var request = require('request');
var util = require('util');

PostManager.prototype.data = {}

PostManager.handleData = function (data) {
  data = JSON.parse(data);
  //console.log(data.data.children);
  var results = [];
  if (data && data.data && data.data.children && data.data.children.length > 0){
    for(var i = 0; i < data.data.children.length; i++){
      var result = data.data.children[i].data;
      if (result){
        results.push({
          date : result.created ? new Date(result.created).toString() : 'NaN',
          postUrl: result.url,
          postTitle: result.title,
          author: result.author,
          postContent: result.selftext,
          postId : result.id
        });
      }
    }
  }

  return results;
}

PostManager.findByTerm = function (term, limit, offset, callback) {
  request("https://www.reddit.com/search.json?q="+term+"&limit="+limit+"&offset=" + offset, function (error, response, body) {
    if (!error && response.statusCode == 200) {
      //console.log() // Show the HTML for the Google homepage.
      console.log("request ok");
      callback(null, PostManager.handleData(response.body));
    }
    else
      callback(error, response);
  });
}

module.exports = PostManager;
